File: app/schemas/task_log.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime, date
from typing import Optional
from enum import Enum

class TaskStatus(str, Enum):
    completed = "completed"
    failed = "failed"
    skipped = "skipped"

class TaskLogCreate(BaseModel):   
    task_id: int
    user_id: int
    log_date: date
    status: TaskStatus
    reason: Optional[str] = None

class TaskLogRead(BaseModel):
    tasklog_id: int
    task_id: int
    user_id: int
    log_date: date
    status: TaskStatus
    reason: Optional[str]
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: web-app/src/types/tasklog.ts
export type TaskStatus = "completed" | "failed" | "skipped";

export interface TaskLogCreate {
  task_id: number;
  log_date: string; // YYYY-MM-DD
  status: TaskStatus;
  reason?: string | null;
}

export interface TaskLogRead {
  tasklog_id: number;
  task_id: number;
  user_id: number;
  log_date: string;
  status: TaskStatus;
  reason: string | null;
  created_at: string;
}
